gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-edit-page/api-documentation-v4-edit-page.component.ts

```diff
@@ -69,19 +69,36 @@ export class ApiDocumentationV4EditPageComponent implements OnInit, OnDestroy {
   }
 
   save() {
+    this.savePage().subscribe(() => {
+      this.ajsState.go('management.apis.documentationV4', { ...this.ajsStateParams, parentId: this.page.parentId });
+    });
+  }
+
+  saveAndPublish() {
+    this.savePage()
+      .pipe(switchMap(() => this.apiDocumentationService.publishDocumentationPage(this.ajsStateParams.apiId, this.ajsStateParams.pageId)))
+      .subscribe({
+        next: () => {
+          this.ajsState.go('management.apis.documentationV4', { ...this.ajsStateParams, parentId: this.page.parentId });
+        },
+        error: (error) => {
+          this.snackBarService.error(error?.error?.message ?? 'Cannot save page');
+        },
+      });
+  }
+
+  private savePage() {
     const editPage: EditDocumentationMarkdown = {
       ...this.page,
       type: 'MARKDOWN',
       content: this.content,
     };
-    this.apiDocumentationService.updateDocumentationPage(this.ajsStateParams.apiId, this.ajsStateParams.pageId, editPage).subscribe({
-      next: () => {
-        this.ajsState.go('management.apis.documentationV4', { ...this.ajsStateParams, parentId: this.page.parentId });
-      },
-      error: (error) => {
-        this.snackBarService.error(error?.error ?? 'Cannot save page');
-      },
-    });
+    return this.apiDocumentationService.updateDocumentationPage(this.ajsStateParams.apiId, this.ajsStateParams.pageId, editPage).pipe(
```

Comment:
Should there be a GET before the PUT for updating a page?

Reply:
Yes good catch